internal/security/setup/option/pkiOption.go

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+package option
+
+import (
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+// OptionsExecutor is an executor to process the input options
+type OptionsExecutor interface {
+	ProcessOptions() (int, error)
+	executeOptions(...func(*PkiInitOption) (exitCode, error)) (exitCode, error)
+}
+
+// PkiInitOption contains command line options for pki-init
+type PkiInitOption struct {
+	LoggingClient logger.LoggingClient
+	executor      OptionsExecutor
+}
+
+// NewPkiInitOption constructor to get options built for pki-init
+func NewPkiInitOption(opts PkiInitOption) (ex OptionsExecutor, statusCode int, err error) {
+	opts.executor = &opts
+
+	if opts.LoggingClient == nil { // use the default one
+		opts.LoggingClient = logger.NewClient("security-secrets-setup", false,
```

Comment:
"security-secrets-setup" string is used in multiple files now. could it be placed and used from the config file?

Reply:
can define a constant for it but not sure about using it from the config file because that name is design time decision instead of run-time configuration.